src/store/user/index.js

```diff
@@ -96,17 +98,16 @@ const actions = {
     const defaultMembershipId = getters['defaultMembershipId']
     const defaultTenant = getters['memberships']?.find(
       membership => membership.id === defaultMembershipId
-    ).tenant
+    )?.tenant
 
     const firstTenant =
-      getters['memberships']?.[0] || rootGetters['tenant/tenants']
-
-    if (!defaultMembershipId || rootGetters['api/isServer']) return firstTenant
+      getters['memberships']?.[0]?.tenant || rootGetters['tenant/tenants']?.[0]
```

Comment:
Really good catches on this, hopefully this'll fix some of those edge cases with missing tenants on load.